### PR TITLE
Added test for the add_subdirectory cmake usage, and added docs

### PR DESCRIPTION
**Pull request by atomgalaxy**
Docs & test for add_subdirectory.

### CODE REVIEW

**Comment by atomgalaxy** [comment]
By popular demand :)...

You needed it, now you have it. I guess it's only fair you review it :)

File: Makefile [hunk 1]
```diff
@@ -1,41 +1,61 @@
...
+
+.PHONY: test_cmake_add_subdir
+test: test_cmake_add_subdir
+test_cmake_add_subdir: $(TEST_DIR)/ciabatta-copy
+
 
+$(TEST_DIR)/cmake-build: CMakeLists.txt test/CMakeLists.txt
+	rm -rf $(TEST_DIR)/cmake-build
+	mkdir -p $(TEST_DIR)/cmake-build
+	cd $(TEST_DIR)/cmake-build && \
```
Comment (timuraudio) [inline Makefile:21]: wait... this test will fail if the user doesn't have `ninja` locally installed, right? `ninja` doesn't come out of the box with Windows/macOS. Should this use MSVC on Windows and Xcode on macOS instead? (and make on Linux)?

Comment (atomgalaxy) [reply Makefile:21]: I guess it's possible to use cmake to invoke the driver, I just haven't figured how yet (should be easy though).
This is automating the testing for deployment, *with a makefile* :). I'm not even sure how one runs it on windows.

Would you agree that generalizing the testing infra is a separate PR?

Comment (timuraudio) [reply Makefile:21]: Sure, that's a separate issue. Feel free to merge this then.

**Pull request merged**